### ISSUE TITLE
GPS not updating on iOS?

### ISSUE THREAD

**Issue by celstark2** [kivy/plyer#173]
I'm new to Plyer and Kivy and have started with the GPS example code in Plyer here.  Getting things to run on the phone is no problem and other examples work just fine (e.g., accelerometer), but the GPS example never seems to have on_location be triggered - at least on iOS 9.  I've hacked the accelerometer example to also have a GPS line and again, my callback for on_location never gets called.

Any ideas / or can anyone else verify this?  The plyer that I've got with kivy-ios seems to be 1.2.5dev

Craig

**Comment by celstark2** [comment]
FWIW, I can confirm that the same code runs just fine on Android now.  None of my attempts to get iOS to fire off the events - provided code or mine - were successful.  There's a .plist entry that needs to be in place (plyer source code notes it - NSLocationAlwaysUsageDescription), but even with that I could not get any GPS events on iOS.

**Comment by gorgonaut04** [comment]
Seeing the same issue. Added the info.plist entry and now I get the popup, but still no luck

**Comment by gorgonaut04** [comment]
So it started working later in the day, and seems to work now... not sure why!

**Comment by doratoa** [comment]
Required key is actually **NSLocationWhenInUseUsageDescription** but I suppose **NSLocationAlwaysUsageDescription** may also work. 
Check my PR for a bit more details: https://github.com/kivy/plyer/pull/187

**Comment by gorgonaut04** [comment]
The key to working with the current code is triggering gps.start() from App.on_resume(), since the app resumes once the user makes a selection. Not sure if there is any code change needed, but some additional documentation would be useful for others.

**Comment by doratoa** [comment]
Yes I didn't think of it at first. Since permission requested only on first run it will work on second run anyway.  This point needs testing and probably documenting.

**Comment by doratoa** [comment]
I've added app.GPS_ACTIVE = False Flag to my App class, add set it to change with gps.start()/stop() (I use gps with mapview only in one Screen)
And this (NOT to be blindly copied, but works like charm for me)
```
    def on_pause(self):
        print("App paused")
        if self.GPS_ACTIVE:
            print("GPS!! Try to stop before pause!")
            try:
                plyer.gps.stop()
            except Exception as exc:
                print("FAILED!: {0}".format(exc))
            else:
                print("GPS!! OK")

    def on_resume(self):
        if self.GPS_ACTIVE:
            print("GPS!! Try to restart on resume!")
            try:
                plyer.gps.start()
            except Exception as exc:
                print("FAILED!: {0}".format(exc))
            else:
                print("GPS!! RESTART OK")
```
